operator/internal/manifests/env_vars.go

```diff
@@ -0,0 +1,85 @@
+package manifests
+
+import (
+	"strings"
+
+	lokiv1 "github.com/grafana/loki/operator/apis/loki/v1"
+	"github.com/imdario/mergo"
+	"github.com/operator-framework/operator-lib/proxy"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	httpProxyKey  = "HTTP_PROXY"
+	httpsProxyKey = "HTTPS_PROXY"
+	noProxyKey    = "NO_PROXY"
+)
+
+func configureProxyEnv(spec *lokiv1.ClusterProxy, pod *corev1.PodSpec) error {
+	if spec == nil {
+		return nil
+	}
+
+	for _, envVar := range proxy.ProxyEnvNames {
+		resetProxyVar(pod, envVar)
+	}
+
+	envVars := proxy.ReadProxyVarsFromEnv()
+	if !spec.ReadVarsFromEnv {
+		envVars = []corev1.EnvVar{
+			{
+				Name:  httpProxyKey,
+				Value: spec.HTTPProxy,
+			},
+			{
+				Name:  strings.ToLower(httpProxyKey),
+				Value: spec.HTTPProxy,
+			},
+			{
+				Name:  httpsProxyKey,
+				Value: spec.HTTPSProxy,
```

Comment:
I think you need a check if the value of a setting is empty and only add the env-var in that case, so that it is possible to set `httpProxy` without also setting `httpsProxy` and `noProxy`. Currently only setting `httpProxy` for example looks like this:

```
        env:
        - name: HTTP_PROXY
          value: test
        - name: http_proxy
          value: test
        - name: HTTPS_PROXY
        - name: https_proxy
        - name: NO_PROXY
        - name: no_proxy
```